refinery/tool_manager/models.py

```diff
@@ -1095,7 +1102,7 @@ def _get_galaxy_history_dataset_list(self):
         ]
         return retained_datasets
 
-    def _get_exposed_galaxy_datasets(self):
+    def _get_exposed_galaxy_datasets(self, exposed_dataset_list=None):
```

Comment:
Is this argument necessary? I only see this function called with `exposed_dataset_list = self._get_galaxy_history_dataset_list()` in `create_analysis_output_node_connections()`

Reply:
it is needed to prevent repeated calls to _get_galaxy_history_dataset_list, which is a big API call.

Reply:
Right but that call is made anyway (https://github.com/refinery-platform/refinery-platform/pull/3461/files#diff-8beee8c9215c07566345b13603f3b9c6R797) and the result is passed as an argument on the next line.

Reply:
ah i see what you are saying - will update

Reply:
ah, @hackdna, i stand corrected.  the AnalysisNodeConnection objects correspond to the output of `_get_galaxy_history_dataset_list` and not `_get_exposed_galaxy_datasets` (which is a subset) - therefore it is necessary actually, as far as i can tell, as both calls need to be made in `create_analysis_output_node_connections`

Reply:
barring the refactor, i would just leave this.

Reply:
I see, in that case I think making this argument required would be a better choice: it would simplify the logic here and it would be a step toward refactoring.

Reply:
if it's all the same to you, i would just leave it.  i am not sure what the refactor is going to look like yet and it might involve changing these functions and their names altogether.  on top of that i would have to fix the tests